Cortex M3 .hex files

The CC2538 and CC26xx ports are identical in terms of how they produce the final .bin and .hex

```
arm-none-eabi-objcopy -O ihex cc2538-demo.elf cc2538-demo.hex
arm-none-eabi-objcopy -O binary --gap-fill 0xff cc2538-demo.elf cc2538-demo.bin
```

In a nutshell, we create a .elf and from it we create a .bin and a .hex

The `.bin` works perfectly with CC2538 as well as with CC26xx. The .hex works fine with CC2538, but it fails miserably with CC26xx. TI's flash prog complains about an unknown record type, but I suspect this is not the primary cause.

I ran objdump on both hex files

```
$ arm-none-eabi-objdump -h cc26xx-demo.hex 

cc26xx-demo.hex:     file format ihex

Sections:
Idx Name          Size      VMA               LMA               File off  Algn
  0 .sec1         0000d6bd  00000000  00000000  00000000  2**0
                  CONTENTS, ALLOC, LOAD
  1 .sec2         00000008  0000d6c0  0000d6c0  00025bf6  2**0
                  CONTENTS, ALLOC, LOAD
  2 .sec3         00000054  0001ffac  0001ffac  00025c24  2**0
                  CONTENTS, ALLOC, LOAD
```

and this looks normal to me.

Any ideas?

I wanted to take a look at it, but the `cc26xx-demo` example does not build using the current instructions and dependencies:
```
  CC        ../../platform/srf06-cc26xx/./contiki-main.c
../../platform/srf06-cc26xx/./contiki-main.c:64:37: fatal error: driverlib/driverlib_ver.h: No such file or directory
 #include "driverlib/driverlib_ver.h"
                                     ^
compilation terminated.
```

`driverlib/driverlib_ver.h` and the `DRIVERLIB_*_VER` definitions no longer exist (new release of cc26xxware on May 5). They seem to have been superseded by `driverlib/driverlib_release.h`. Older releases are not available for download, and anyway the documentation in Contiki does not say which version it is supposed to be compatible with.

Contiki should not rely on an external dependency out of control. The supported archives should be either made available for download on some servers controlled by Contiki (at least as a backup), or the source code they contain should be integrated into Contiki and regularly upgraded, or this could also be a Git submodule under the contiki-os organization. And all these operations could be automated for the build instead of having to manually download this dependency and set `TI_CC26XXWARE`.

It's not the 0x04 record type that CC2538 is using instead of 0x03 on CC26xx, but 0x05. 0x04 is just used to program stuff at 0x20XXXX, then at 0x27XXXX. The issue with CC26xx is that it's using segmented addresses instead of linear addresses.

I've removed the offending version stuff, just to try building, but then it breaks everywhere, e.g.:
```
In file included from ../../platform/srf06-cc26xx/srf06/button-sensor.c:46:0:
../../platform/srf06-cc26xx/srf06/button-sensor.c: In function 'config_buttons':
../../platform/srf06-cc26xx/srf06/button-sensor.c:61:34: error: 'IOC_JTAG_DISABLE' undeclared (first use in this function)
                                  IOC_JTAG_DISABLE)
                                  ^
../../cpu/cc26xx/./ti-lib.h:256:66: note: in definition of macro 'ti_lib_ioc_port_configure_set'
 #define ti_lib_ioc_port_configure_set(...)   IOCPortConfigureSet(__VA_ARGS__)
                                                                  ^
../../platform/srf06-cc26xx/srf06/button-sensor.c:198:55: note: in expansion of macro 'BUTTON_GPIO_CFG'
     ti_lib_ioc_port_configure_set(key, IOC_PORT_GPIO, BUTTON_GPIO_CFG);
                                                       ^
../../platform/srf06-cc26xx/srf06/button-sensor.c:61:34: note: each undeclared identifier is reported only once for each function it appears in
                                  IOC_JTAG_DISABLE)
                                  ^
../../cpu/cc26xx/./ti-lib.h:256:66: note: in definition of macro 'ti_lib_ioc_port_configure_set'
 #define ti_lib_ioc_port_configure_set(...)   IOCPortConfigureSet(__VA_ARGS__)
                                                                  ^
../../platform/srf06-cc26xx/srf06/button-sensor.c:198:55: note: in expansion of macro 'BUTTON_GPIO_CFG'
     ti_lib_ioc_port_configure_set(key, IOC_PORT_GPIO, BUTTON_GPIO_CFG);
                                                       ^
../../Makefile.include:241: recipe for target 'obj_srf06-cc26xx/button-sensor.o' failed
```

Very incompatible new cc26xxware release...

--- a/.travis.yml
+++ b/.travis.yml
@@ -41,10 +41,11 @@ before_script:
 
   ## Install mainline ARM toolchain.  gcc-arm-none-eabi is available
   ## in Ubuntu >= 14.04, but this external PPA is needed for 12.04.
+  ## Install srecord
   - if [ ${BUILD_ARCH:-0} = arm-aapcs ] ; then
       sudo add-apt-repository -y ppa:terry.guo/gcc-arm-embedded &&
       sudo apt-get -qq update &&
-      sudo apt-get -qq install gcc-arm-none-eabi &&
+      sudo apt-get -qq install gcc-arm-none-eabi srecord &&
       arm-none-eabi-gcc --version ;
     fi
 

--- a/cpu/cc26xx/Makefile.cc26xx
+++ b/cpu/cc26xx/Makefile.cc26xx
@@ -6,6 +6,7 @@ OBJCOPY = arm-none-eabi-objcopy
 OBJDUMP = arm-none-eabi-objdump
 NM      = arm-none-eabi-nm
 SIZE    = arm-none-eabi-size
+SREC_CAT = srec_cat
 
 ### TI CC26xxware out-of-tree
 ### TI_CC26XXWARE is the home directory of the cc26xxware
@@ -106,8 +107,11 @@ CUSTOM_RULE_LINK=1
 	$(TRACE_LD)
 	$(Q)$(LD) $(LDFLAGS) ${filter-out $(LDSCRIPT) %.a,$^} ${filter %.a,$^} $(TARGET_LIBFILES) -lm -o $@
 
-%.hex: %.elf
+%.i16hex: %.elf
 	$(OBJCOPY) -O ihex $< $@
+
+%.hex: %.i16hex
+	$(SREC_CAT) $< -intel -o $@ -intel
 
 %.bin: %.elf
 	$(OBJCOPY) $(OBJCOPY_FLAGS) $< $@

--- a/platform/srf06-cc26xx/README.md
+++ b/platform/srf06-cc26xx/README.md
@@ -69,6 +69,7 @@ To use the port you need:
     [...]
     gcc version 4.9.3 20141119 (release) [ARM/embedded-4_9-branch revision 218278] (GNU Tools for ARM Embedded Processors)
 
+* srecord (http://srecord.sourceforge.net/)
 * You may also need other drivers so that the SmartRF can communicate with your
 operating system and so that you can use the chip's UART for I/O. Please read
 the section ["Drivers" in the CC2538DK readme](https://github.com/contiki-os/contiki/tree/master/platform/cc2538dk#drivers).

